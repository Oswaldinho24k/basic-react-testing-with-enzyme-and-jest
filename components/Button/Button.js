import PropTypes from 'prop-types';

/**
 * Component for showing details of the user.
 * 
 * @component
 * @example
 * 
 * const handleClick = () => console.log('clicked')
 * return (
 *   <Button disabled={false} onClick={handleClick}> The Button </Button>
 * )
 */

const Button = ({children, disabled, onClick}) => {
  return(
    <button disabled={disabled} onClick={onClick}>
      {children}
    </button>
  )
}

Button.defaultProps = {
  disabled: false
}

Button.propTypes = {
  /**
   * Buttons's disabled status
   */
  disabled: PropTypes.bool,
  /**
   * Buttons's onClick function
   */
  onClick: PropTypes.func,
  /**
   * Buttons's children text
   */
  children: PropTypes.any.isRequired
}

export default Button
