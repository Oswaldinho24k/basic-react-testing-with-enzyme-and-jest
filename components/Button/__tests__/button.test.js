import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import Button from "../Button";

describe("Render Button based on props", () => {
  const handleClick = jest.fn()
  test("if disabled prop, button is disabled", () => {
    const button = renderer.create(<Button disabled={true} onClick={handleClick}>text</Button>)
    const tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("if disabled == false, button can be clickable", () => {
    const button = renderer.create(<Button onClick={handleClick}>text</Button>)
    const tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  })
})