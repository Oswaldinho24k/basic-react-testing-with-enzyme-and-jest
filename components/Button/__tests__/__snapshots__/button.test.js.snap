// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button based on props if disabled == false, button can be clickable 1`] = `
<button
  disabled={false}
  onClick={[MockFunction]}
>
  text
</button>
`;

exports[`Render Button based on props if disabled prop, button is disabled 1`] = `
<button
  disabled={true}
  onClick={[MockFunction]}
>
  text
</button>
`;
